src/cbor.c

```diff
@@ -105,14 +105,14 @@ void cbor_advanceToken(stream_t* stream)
 }
 
 
-size_t cbor_writeToken(uint8_t type, uint64_t value, uint8_t *buf, size_t bufSize)
+size_t cbor_writeToken(uint8_t type, uint64_t value, uint8_t* inBuffer, size_t inSize)
 {
-	ASSERT(bufSize < BUFFER_SIZE_PARANOIA);
+	ASSERT(inSize < BUFFER_SIZE_PARANOIA);
 
-#define CHECK_BUF_LEN(requiredSize) if (requiredSize > bufSize) THROW(ERR_DATA_TOO_LARGE);
+#define CHECK_BUF_LEN(requiredSize) if (requiredSize > inSize) THROW(ERR_DATA_TOO_LARGE);
```

Comment:
Hmm, maybe we should have `if ((size_t) requiredSize > inSize)` just to make sure an accidental `-1` wouldn't pass the test